src/ImportDefinitionsBundle/Export/Exporter.php

```diff
@@ -0,0 +1,266 @@
+
+                    $this->exceptions[] = $ex;
+
+                    $this->eventDispatcher->dispatch(
+                        'export_definition.status',
+                        new ImportDefinitionEvent($definition, sprintf('Error: %s', $ex->getMessage()))
+                    );
+
+                    if ($definition->getStopOnException()) {
+                        throw $ex;
+                    }
+                }
+
+                $this->eventDispatcher->dispatch(
+                    'export_definition.progress',
+                    new ImportDefinitionEvent($definition)
+                );
+            }
+        }
+
+        print_r($entries);
+    }
+
+    /**
+     * @param DefinitionInterface $definition
+     * @param Concrete $object
+     * @param $params
+     * @return null|Concrete
+     * @throws \Exception
+     */
+    private function exportRow(DefinitionInterface $definition, Concrete $object, $params): array
+    {
+        $data = [];
+
+        $runner = null;
+
+        $this->eventDispatcher->dispatch('export_definition.status', new ImportDefinitionEvent($definition, sprintf('Export Object %s', $object->getId())));
+        $this->eventDispatcher->dispatch('export_definition.object.start', new ImportDefinitionEvent($definition, $object));
+
+        if ($definition->getRunner()) {
+            $runner = $this->runnerRegistry->get($definition->getRunner());
+        }
+
+        if ($runner instanceof ExportRunnerInterface) {
+            $data = $runner->exportPreRun($object, $data, $definition, $params);
+        }
+
+        $this->logger->info(sprintf('Export Object: %s', $object->getRealFullPath()));
+
+        foreach ($definition->getMapping() as $mapItem)
+        {
+            $data[$mapItem->getFromColumn()] = $this->getObjectValue($object, $mapItem, $data, $definition, $params, $runner);
+        }
+
+        $this->eventDispatcher->dispatch('export_definition.status', new ImportDefinitionEvent($definition, sprintf('Exported Object %s', $object->getFullPath())));
+        $this->eventDispatcher->dispatch('export_definition.object.finished', new ImportDefinitionEvent($definition, $object));
+
+        if ($runner instanceof ExportRunnerInterface) {
+            $data = $runner->exportPostRun($object, $data, $definition, $params);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param Concrete             $object
+     * @param Mapping              $map
+     * @param                      $data
+     * @param DefinitionInterface  $definition
+     * @param                      $params
+     * @param RunnerInterface|null $runner
+     * @return mixed|null
+     * @throws DoNotSetException
+     */
+    private function getObjectValue(Concrete $object, Mapping $map, $data, DefinitionInterface $definition, $params, RunnerInterface $runner = null)
+    {
+        $value = null;
+
+        if ($map->getGetter()) {
+            $setter = $this->getterRegistry->get($map->getSetter());
+
```

Comment:
Is this a mistake? Is it not supposed to be $map->getGetter() you try to get from getterRepository? 

Also I saw that the setter for localized fields also implement GetterInterface. Maybe add an elseif here so if there is no getter defined in map then try to use setter instead if the setter implements the GetterInterface?

Reply:
see https://github.com/w-vision/ImportDefinitions/pull/97, that is right. All setters also implement the getter, since the functionality is the same just reversed. But they have there on registry and are registered separately to the container.